composer/core/state.py

```diff
@@ -907,13 +907,19 @@ def state_dict(self) -> Dict[str, Any]:
                     v.persistent(mode=True)
                     # We cast the metric tensor to a numpy array, so that FSDP doesn't mistake it for a tensor to be sharded upon load.
                     _computed = v._computed
-                    _computed_device = str(_computed.device) if _computed is not None else None
-                    _np_computed = _computed.cpu().numpy() if _computed is not None else None
-                    serialized_value[k] = {
-                        'state_dict': v.state_dict(),
-                        '_computed': _np_computed,
-                        '_computed_device': _computed_device
-                    }
+                    if isinstance(_computed, torch.Tensor):
+                        _computed_device = str(_computed.device) if _computed is not None else None
+                        _np_computed = _computed.cpu().numpy() if _computed is not None else None
+                        serialized_value[k] = {
+                            'state_dict': v.state_dict(),
+                            '_computed': _np_computed,
+                            '_computed_device': _computed_device
+                        }
+                    else:
+                        serialized_value[k] = {
```

Comment:
what is it is not a tensor for the else case/are we confident the else path works?

Reply:
@eracah has said they only do weird invasive things if it is a single tensor. I have added a MAP test which verifies this (it has a dict)